lib/iris/exceptions.py

```diff
@@ -35,6 +35,12 @@ class AncillaryVariableNotFoundError(KeyError):
     pass
 
 
+class CubeComponentNotFoundError(KeyError):
+    """Raised when a search yields no cube components."""
+
+    pass
+
+
```

Comment:
Since we'll now have this general error, will the more specific ones even be needed? This feels pretty specific as it is.

Reply:
I think I'll push for them to inherit from this, it just makes sense and breaks nothing. Feel free to push back.